repository_service_tuf_worker/repository.py

```diff
@@ -308,29 +315,42 @@ def _bump_and_persist(
         if persist:
             self._persist(role, role_name)
 
-    def _update_timestamp(self, snapshot_version: int) -> Metadata[Timestamp]:
+    def _update_timestamp(
+        self,
+        snapshot_version: Optional[int] = None,
+        skip: Optional[bool] = False,
+    ) -> Metadata[Timestamp]:
         """
         Loads 'timestamp', updates meta info about passed 'snapshot'
         metadata, bumps version and expiration, signs and persists.
 
         Args:
-            snapshot_version: snapshot version to add to new timestamp.
-            db_targets: RSTUFTarget DB objects will be changed as published in
-                the DB SQL.
+            snapshot_version: Optional snapshot version to new timestamp.
+            skip: Skip bumping version and expiration if snpashot
+            version is not provided
         """
         timestamp: Metadata[Timestamp] = self._storage_backend.get(
             Timestamp.type, None
         )
+        if not snapshot_version and skip:
+            logging.debug(
+                "No snapshot version and skip is True. Not bumping Timestamp"
+            )
+            return timestamp
+
         timestamp.signed.snapshot_meta = MetaFile(version=snapshot_version)
 
         self._bump_and_persist(timestamp, Timestamp.type)
 
+        logging.debug("Bumped version of 'Target' role")
         return timestamp
 
     def _update_snapshot(
         self,
         target_roles: Optional[List[str]] = None,
         bump_all: Optional[bool] = False,
+        only_target: Optional[bool] = False,
+        only_snapshot: Optional[bool] = False,
```

Comment:
You can simplify this function just by removing `only_snapshot`.

The previous implementation of `bump_snapshot(force=True)` did exactly what you want to achieve with this argument - it called `update_timestamp(update_snapshot())` without providing any argument.
This meant that `target_roles` or `bump_all` were not provided and it only bumped snapshot.

Maybe you can rename `bump_snapshot` to `bump_only_snapshot` or `bump_snapshot_without_targets()` or something like this and you will be able to remove `only_snapshot`.

Reply:
I favor a full refactoring of this function but outside this PR.
I think we can add these comments in the same issue (with the https://github.com/repository-service-tuf/repository-service-tuf-worker/pull/593#discussion_r1753162614)

Reply:
Okay, as we are limited on time, lets do it in another pr after the conference.